src/vorta/views/partials/treemodel.py

```diff
@@ -994,3 +995,32 @@ def lessThan(self, left: QModelIndex, right: QModelIndex) -> bool:
         data1 = self.choose_data(left)
         data2 = self.choose_data(right)
         return data1 < data2
+
+    def setFilterFixedString(self, pattern: str):
+        """
+        Set the pattern to filter for.
+        """
+        self.searchPattern = pattern
+        self.invalidateRowsFilter()
```

Comment:
Have you read the notice that `beginResetModel` might be more efficient?

Reply:
I had not, thank you I'll check it.

Reply:
Last paragraph of https://doc.qt.io/qt-6/qsortfilterproxymodel.html#filtering

Reply:
I'm not sure if it will be actually more efficient. It does mention it will require model to be repopulated and I want to invoke search through a button so it won't actually be called a lot of times (depends on the users though).
Let's leave this review open for now and once all filters have been added, I'll try running some basic tests to see which one performs better.

Reply:
I think resetting model will also loose the checkbox state in extract view. So if someone needs to extract files from multiple searches, it wouldn't work.

Reply:
Ok, that rules out resetting the model.